### PR TITLE
fix(mongodb): use $or + $eq instead of $in

### PR DESCRIPTION
**Pull request by laplab**
Closes https://github.com/prisma/team-orm/issues/1119

### CODE REVIEW

**Comment by laplab** [comment]
I checked all the other places where we use `$in` operator as well. In all the remaining cases we:
1. Either cannot replace `$in` operator with something else. For example, the query requires us to check for existence of an element in an array stored in a field.
2. Or it does not make sense to replace it because it is actually an MQL flavour of `$in` operator which MongoDB optimises for index usage

File: query-engine/connectors/mongodb-query-connector/src/filter.rs [hunk 1]
```diff
@@ -188,42 +188,53 @@ impl MongoFilterVisitor {
...
                         }
-
-                        doc! { "$in": [&field_name, bson_values] }
-                    }
-                    _ => {
-                        doc! { "$in": [&field_name, self.coerce_to_bson_for_filter(field, PrismaValue::List(vals))?] }
                     }
-                },
+
+                    doc! { "$or": equalities }
```
Comment (Weakky) [inline query-engine/connectors/mongodb-query-connector/src/filter.rs:217]: Could you perhaps add a small comment that retains the info on why we switched to `$or`?

**Review by Weakky — APPROVED** [review]
Nice 👍🏻

**Comment by laplab** [comment]
Clippy failure is unrelated and is getting fixed in https://github.com/prisma/prisma-engines/pull/4849.

**Pull request merged**